wicketd/tests/integration_tests/updates.rs

```diff
@@ -72,12 +73,31 @@ async fn test_updates() {
         .await
         .expect("update started successfully");
 
-    let _status = wicketd_testctx
-        .wicketd_client
-        .get_update_sp(wicketd_client::types::SpType::Sled, 0)
-        .await
-        .expect("get_update successful")
-        .into_inner();
+    let terminal_event = 'outer: loop {
+        let status = wicketd_testctx
+            .wicketd_client
+            .get_update_sp(wicketd_client::types::SpType::Sled, 0)
+            .await
+            .expect("get_update_sp successful")
+            .into_inner();
+
+        for event in status.events {
+            if let UpdateEventKind::Terminal(event) = event.kind {
+                break 'outer event;
+            }
+        }
+    };
+
+    match terminal_event {
+        UpdateTerminalEventKind::ArtifactUpdateFailed { artifact, .. } => {
+            // TODO: obviously we shouldn't stop here, get past more of the
+            // update process in this test.
```

Comment:
I've set this up and it's been a lot of work, but also I don't feel like going further down this path is the highest priority for now.

There's a lot of work that needs to be done here to make this test behave more like the real system. For example, the test currently fails with "updating SP failed: failed to parse SP update as a zip file: invalid Zip archive: Could not find central directory end", because the fake SP artifact doesn't look like a zip file.

Maybe we can make this test use a real artifact downloaded over HTTP or similar, or extend the fake artifact to work for this, but for now I think we should just move on and instead rely on manual tests with real systems. What do you think?

An outcome that might be worth pursuing is creating a binary that runs all of wicketd, MGS and sp-sim within the same process, so the full flow is easier to test.

Reply:
Agree with both of you

Reply:
Thanks both of you -- this was a good learning experience but I'm glad I was able to timebox it and move on.